app/frontend/src/common/store/documents.js

```diff
@@ -111,7 +114,23 @@ export default {
       state.file_upload_success = payload
     },
     setFiles (state, payload) {
-      state.upload_files = payload
+      if (payload.length > 0) {
```

Comment:
Simpler? One line replacing everything?
```suggestion
      state.upload_files.push(...payload)
```